### DISCUSSION TITLE
No sound suddenly -- everything appears normal, nothing changed

### DISCUSSION THREAD

**Discussion by chrisbenincasa** [dtcooper/raspotify#656] [Q&A] [answered]
Today, after changing nothing, I was suddenly unable to get any sound output from Raspotify. I'm not sure what might have changed... here's what I've tried so far and some facts about my system:

Facts:
```
Linux apollo 6.1.21-v8+ #1642 SMP PREEMPT Mon Apr  3 17:24:16 BST 2023 aarch64 GNU/Linux
PRETTY_NAME="Debian GNU/Linux 11 (bullseye)"
NAME="Debian GNU/Linux"
VERSION_ID="11"
VERSION="11 (bullseye)"
VERSION_CODENAME=bullseye
ID=debian
```

<details>
<summary>`-- Ouput of aplay -l` --</summary>

```
**** List of PLAYBACK Hardware Devices ****
card 0: sndrpihifiberry [snd_rpi_hifiberry_digi], device 0: HifiBerry Digi HiFi wm8804-spdif-0 [HifiBerry Digi HiFi wm8804-spdif-0]
  Subdevices: 1/1
  Subdevice #0: subdevice #0
```
</details>

<details>

<summary>`-- Ouput of aplay -L --`</summary>
```
null
    Discard all samples (playback) or generate zero samples (capture)
iec958
    IEC958 (S/PDIF) Digital Audio Output
default
hw:CARD=sndrpihifiberry,DEV=0
    snd_rpi_hifiberry_digi, HifiBerry Digi HiFi wm8804-spdif-0
    Direct hardware device without any conversions
plughw:CARD=sndrpihifiberry,DEV=0
    snd_rpi_hifiberry_digi, HifiBerry Digi HiFi wm8804-spdif-0
    Hardware device with all software conversions
sysdefault:CARD=sndrpihifiberry
    snd_rpi_hifiberry_digi, HifiBerry Digi HiFi wm8804-spdif-0
    Default Audio Device
iec958:CARD=sndrpihifiberry,DEV=0
    snd_rpi_hifiberry_digi, HifiBerry Digi HiFi wm8804-spdif-0
    IEC958 (S/PDIF) Digital Audio Output
dmix:CARD=sndrpihifiberry,DEV=0
    snd_rpi_hifiberry_digi, HifiBerry Digi HiFi wm8804-spdif-0
    Direct sample mixing device
```
</details>

Things I've tried
* System restart
* Reinstalling Raspotify
* Connecting from various clients (iPhone, iPad, etc)

Things I've determined:
* I don't think this is an audio device issue because:
  * I also run Plexamp headless on this device which works flawlessly
  * I manually started a separate instance of librespot with `librespot -n "HelloWorld" -b 160 -v`, connected, and audio was played
* I don't think this is an mDNS / discovery issue because of my success with running librespot directly (and seeing similar unreadable packet warnings in that run)
* Verbose logs look normal from Raspotify when "playing" music, just no output at all...

Any help would be greatly appreciated!!! Love Raspotify and would really love to get it working again.

**Comment by micmat00** [comment]
Did you manage to solve the problem? I'm currently facing the same issue. It stopped working overnight and the following error message is displayed:
-Track should be available, but no alternatives found.
-<spotify:track:2QTeQ2tJcuv5NNvlbkjxwR> is not available
-Skipping to next track, unable to load track <SpotifyId("spotify:track:2QTeQ2tJcuv5NNvlbkjxwR")>

**Reply by chrisbenincasa** [reply]
Somehow after I originally posted I got things working again... until a day or two ago when similar things started happening. I don't think it's on raspotify but rather librespot. I stumbled on this issue and the /etc/hosts solution at the bottom got things working for me again https://github.com/librespot-org/librespot/issues/1527

Hopefully this helps!

**Reply by micmat00** [reply]
Thank you, it worked for me too!

**Comment by kimtore** [comment, marked as answer]
0.46.2 fixes this issue.
